src/main/groovy/com/ullink/AssemblyInfoVersionPatcher.groovy

```diff
@@ -13,7 +13,7 @@ class AssemblyInfoVersionPatcher extends ConventionTask {
     List<String> projects = []
 
     AssemblyInfoVersionPatcher() {
-        conventionMapping.map "projects", { [ project.tasks.msbuild.mainProject?.projectName ] }
+        conventionMapping.map "projects", { project.tasks.msbuild.projects.collect { it.name } }
```

Comment:
Having mixed feelings about this one.

Maybe you're right and test projects shouldn't even have a version (isn't the attribute mandatory ? it must be for SN'ed projects at least).

For other projects, either they're packaged in an installer and should have the same version as the exe, or they repack'ed and the version doesn't matter because it'll be dropped.